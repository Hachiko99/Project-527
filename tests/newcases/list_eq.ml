type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

type bool =
  | True
  | False

let rec nat_equal (n1 : nat) (n2 :nat) : bool =
  match n1 with
  | O -> (match n2 with
             | O -> True
             | S (m) -> False
           )
  | S (m1) -> ( match n2 with
        | O -> False
        | S (m2) -> (nat_equal m1 m2) )
;;

let list_eq : list -> list -> bool |>
 {
    [] => ( [] => True
          | [0] => False)
  | [2] => ( [] => False
           | [1] => False
           | [2] => True)
  | [1] => ( [] => False
           | [1] => True
           | [2] => False)
  | [1;0] => ( [] => False
             | [1] => False
             | [2;0] => False
             | [1;0] => True)
  | [2;0] => ( [] => False
             | [1] => False
             | [2;0] => True
             | [1;0] => False)
(*  | [2;1;0] => ( [1;0] => False
               | [2;3;0] => False
               | [2;1;0] => True)
  | [1;3;2;0] => ( [3;2;0] => False
                 | [1;3;2;0] => True)*)} = ?
