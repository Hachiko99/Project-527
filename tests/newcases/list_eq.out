(* Generated Tue Apr 19 12:38:25 2016 *)
let list_eq : list -> list -> bool =
  let rec f1 (l1:list) : list -> bool =
    fun (l2:list) ->
      match l2 with
        | Nil -> (match l1 with
                    | Nil -> True
                    | Cons (n1, l3) -> False)
        | Cons (n1, l3) -> (match l1 with
                              | Nil -> False
                              | Cons (n2, l4) -> (match f1 l3 l4 with
                                                    | True -> nat_equal n2 n1
                                                    | False -> False))
  in
    f1
;;
